zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -592,7 +592,7 @@ object HttpCodec extends ContentCodecs with HeaderCodecs with MethodCodecs with
     def index(index: Int): ContentStream[A] = copy(index = index)
   }
   private[http] final case class Query[A](name: String, textCodec: TextCodec[A], index: Int = 0)
-      extends Atom[HttpCodecType.Query, A]  {
+      extends Atom[HttpCodecType.Query, NonEmptyChunk[A]] {
```

Comment:
I would not do `NonEmptyChunk`, because it is valid query string: `foo&bar&baz`. In such case, the chunk will be empty. The RFC explicitly allows this.

Reply:
If we're talking about query codec for `String` values, It seems that currently it work a little bit differently. If query string contains expected query parameter field which doesn't have associated value, this query parameter will be parsed as an empty string. So even with `NonEmptyChunk` we get non empty chunk with empty strings.

For query parameters of other types such query(`foo&bar&baz`) string doesn't work.

Reply:
- If there is a name, but no value, then it should produce empty chunk
 - If there is no name, it should not produce an entry
 - If there is a name and one or more values, it should produce a non empty chunk

This allows us to (1) fail when it's missing, (2) succeed when it's present but no values, (3) fail on more than one values, etc.

Reply:
@jdegoes Please note that failing when single value parameter actually receives multiple query values will break compatibility. Now endpoints successfully handle requests with multiple query values- they just get the first value

Reply:
@ekhov it is okay to break compatibility at this point. We explicitly do not guarantee it before 3.0.0 final. And to get a cleaner API, that is okay at this point.